api/api/uri_parser.py

```diff
@@ -0,0 +1,43 @@
+import functools
+
+import connexion
+
+from api.api_exception import APIException
+from api.util import parse_api_param, raise_if_exc
+
+
+class APIUriParser(connexion.decorators.uri_parsing.OpenAPIURIParser):
+    def __call__(self, function):
+        """
+        :type function: types.FunctionType
+        :rtype: types.FunctionType
+        """
+
+        @functools.wraps(function)
+        def wrapper(request):
+            def coerce_dict(md):
+                """ MultiDict -> dict of lists
+                """
+                try:
+                    return md.to_dict(flat=False)
+                except AttributeError:
+                    return dict(md.items())
+
+            # Raise exception if semicolon is used in q parameter
+            if 'q' in request.query.keys():
+                q = parse_api_param(request.url, 'q')
+                if q:
+                    if ';' in q:
+                        raise_if_exc(APIException(code=2009,  details='Semicolon (;) must be percent-encoded (%3B)'))
```

Comment:
Why do we have to add details here? It is already specified in the exception code 2009